Core/Src/geneva.c

```diff
@@ -88,15 +95,19 @@ int geneva_GetPWM(){
 	return pwm;
 }
 
+float geneva_GetI(){
+	return genevaK.I;
+}
+
 ///////////////////////////////////////////////////// PRIVATE FUNCTION IMPLEMENTATIONS
 
 static void CheckIfStuck(){
 	static uint tick = 0xFFFF;			//
 	static int enc = 0;
-	if(fabs(geneva_Encodervalue() - enc) < ENCODER_DEVIATION_MARGIN){
+	if(fabs(geneva_Encodervalue() - enc) > ENCODER_DEVIATION_MARGIN){

```

Comment:
how does that work?

Reply:
If the value is not the same as the previous encoder value, it is moving, so the tick is the current time. Else if 500 ticks have passed you are stuck.